settings.gradle.kts

```diff
@@ -192,7 +192,7 @@ fun versionRestrictions(
   if (addedRestrictions.isEmpty()) {
     return currentRestrictions
   }
-  return currentRestrictions.union(addedRestrictions)
+  return currentRestrictions.intersect(addedRestrictions)
```

Comment:
the change seems correct if "restrictions" is understood as "supported versions"...
thus we combine sets of supported versions by using their intersection.

by combining 2 sets of restrictions by intersecting we'll have "less restrictions" (potentially no restrictions), which from the current naming seems wrong/confusing imo.